src/app/Http/Controllers/Auth/UserController.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Http\Controllers\Auth;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\UserRequest;
+use App\Models\User;
+use Illuminate\Contracts\View\View;
+use Illuminate\Support\Facades\Hash;
+
+/**
+ * UserController
+ */
+class UserController extends Controller
+{
+    /**
+     * Constructor
+     *
+     * @return void
+     */
+    public function __construct (
+        public User $user,
+    ) {}
+    
+    /**
+     * ユーザ登録画面表示
+     *
+     * @return View
+     */ 
+    public function showRegister(): View
+    {
+        return view('auth.user-register');
+    }
+
+    /**
+     * ユーザ登録処理
+     *
+     * @param UserRequest $request
+     * @return View
+     */
+    public function executeRegister(UserRequest $request): View
+    {
+        $values = $request->merge(['password' => Hash::make($request->password)]);
+
+        $this->user->create($values->only(['name', 'email', 'password',]));
```

Comment:
書き直さなくてもいいけど、パフォーマンス的にはこっちの方が良いかな！
（自分がこの書き方を教えたから書いてくれていると思うけど）

```suggestion
        $this->user->create([
              'name' => $request->name,
              'email' => $request->email,
              'password' => Hash::make($request->password)
        ]);
```

Reply:
▼ `merge()`の詳細 ▼
https://github.com/laravel/framework/blob/6c83a59c1c5e8a91d37c5e7501ea75ad28f5e13d/src/Illuminate/Http/Request.php#L342-L353

▼ `only()`の詳細 ▼
https://github.com/laravel/framework/blob/9.x/src/Illuminate/Http/Concerns/InteractsWithInput.php#L376-L399